src/target/rp.c

```diff
@@ -385,10 +389,117 @@ static void rp_add_flash(target *t, uint32_t addr, size_t length)
         f->blocksize = 0x1000;
         f->erase = rp_flash_erase;
         f->write = rp_flash_write;
-        f->buf_size = 2048; /* Max buffer size used eotherwise */
+        f->buf_size = 2048; /* Max buffer size used otherwise */
+		f->erased = 0xFF;
         target_add_flash(t, f);
 }
 
+void rp_ssel_active(target *t, bool active)
+{
+	const uint32_t qspi_ctrl_outover_low  = 2UL << 8;
+	const uint32_t qspi_ctrl_outover_high = 3UL << 8;
+	uint32_t state = (active) ? qspi_ctrl_outover_low : qspi_ctrl_outover_high;
+	uint32_t val = target_mem_read32(t, QSPI_CTRL_ADDR);
+	val = (val & ~qspi_ctrl_outover_high) | state;
+	target_mem_write32(t, QSPI_CTRL_ADDR, val);
+}
+
+uint32_t rp_read_flash_chip(target *t, uint32_t cmd)
+{
+	uint32_t value = 0;
+
+	rp_ssel_active(t, true);
+
+	/* write command into SPI peripheral's FIFO */
+	for (int count = 0; (count < 4); count++)
```

Comment:
no need for the extra () here as the comparison binds tighter than the ;'s do. Likewise size_t rather than int - no point having this be signed.